src/cljs/org/broadinstitute/firecloud_ui/utils.cljs

```diff
@@ -84,10 +85,26 @@
    (local-storage-write ::use-live-data? ns true)))
 
 
+(defonce ^:private user-listeners (atom {}))
+(defn add-user-listener [k on-change]
+  (swap! user-listeners assoc k on-change))
+(defn remove-user-listener [k]
+  (swap! user-listeners dissoc k))
+
+
 (defonce google-auth2-instance (atom nil))
+(defn set-google-auth2-instance! [instance]
+  (reset! google-auth2-instance instance)
+  (-> instance
+      (aget "currentUser")
+      (js-invoke
+       "listen" (fn [u]
+                  (doseq [[_ on-change] @user-listeners]
+                    (on-change u))))))
 
 (defn get-access-token []
-  (-> @google-auth2-instance (.-currentUser) (.get) (.getAuthResponse) (.-access_token)))
+  (-> @google-auth2-instance
+      (aget "currentUser") (js-invoke "get") (js-invoke "getAuthResponse") (aget "access_token")))
```

Comment:
This pattern is necessary to support advanced compilation. I had to turn that off a while back, but I'd like to collect more data before we decide to abandon it.